Assets/Scripts/AudioScript.cs

```diff
@@ -1,30 +1,78 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using TMPro;
 
 public class AudioScript : MonoBehaviour
 {
     [SerializeField] private AudioSource narrationPlayer;
     [SerializeField] private AudioSource musicPlayer;
 
+    [SerializeField] private TextMeshProUGUI textMesh;
+
+    private AudioTrigger currentTrigger;
+    private bool isNarrativePlaying = false;
+    private float startTime;
+    private int subtitleCounter = 0;
+
     private void OnTriggerEnter(Collider other)
     {
-        if(other.GetComponent<AudioTrigger>() != null)
-        { 
-            if(other.GetComponent<AudioTrigger>().isMusicTrigger)
+        currentTrigger = other.GetComponent<AudioTrigger>();
+        if (currentTrigger != null)
+        {
+            if (currentTrigger.isMusicTrigger)
             {
                 //Debug.Log("Triggered music trigger");
-                musicPlayer.clip = other.GetComponent<AudioTrigger>().triggerClip;
+                musicPlayer.clip = currentTrigger.triggerClip;
                 musicPlayer.Play();
             }
             else
             {
                 //Debug.Log("Triggered audio trigger");
-                narrationPlayer.clip = other.GetComponent<AudioTrigger>().triggerClip;
+                PrepareNarrative();
                 narrationPlayer.Play();
             }
-            if(other.GetComponent<AudioTrigger>().destroyOnPlay)
+            if (currentTrigger.destroyOnPlay)
                 Destroy(other);
         }
     }
+
+    void FixedUpdate()
+    {
+        if (currentTrigger != null && currentTrigger.playSubtitles && isNarrativePlaying)
+        {
+            if (IsSubtitleOver(currentTrigger.subtitleTime[subtitleCounter]))
+            {
+                subtitleCounter++;
+                SetTime();
+            }
+            
+            if (subtitleCounter >= currentTrigger.subtitleText.Length)
+            {
+                isNarrativePlaying = false;
+                textMesh.text = " ";
+            } else
+            {
+                textMesh.text = currentTrigger.subtitleText[subtitleCounter];
+            }
+            
+        }
+    }
+
+    private void PrepareNarrative()
+    {
+        narrationPlayer.clip = currentTrigger.triggerClip;
+        SetTime();
+        isNarrativePlaying = true;
+    }
+
+    private void SetTime()
+    {
+        startTime = Time.time;
+    }
+
+    private bool IsSubtitleOver(float subtitleTime)
+    {
+        return startTime + subtitleTime <= Time.time;
```

Comment:
can you please not use this fancy way of writing it? No need for everyone to learn new syntax like <= here

Reply:
this is just logic math operation, which compares if startTime+ subtitleTime is less or equal to the System Time and returns either true or false 😅 doesn't seem fancy at all, since it's just basic

Reply:
It was done here to separate the concerns in the main place, where it's called, so it doesn't cramp up 'if' sentence with additions, `Time.time` calls and such